src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/idam/cache/IdamCachedClient.java

```diff
@@ -24,68 +21,45 @@
-    private static Cache<String, UserDetails> userDetailsCache;
+    private static Cache<String, CachedIdamCredential> idamCache;
 
     private final IdamClient idamClient;
     private final JurisdictionToUserMapping users;
 
     public IdamCachedClient(
         IdamClient idamClient,
         JurisdictionToUserMapping users,
-        AccessTokenCacheExpiry accessTokenCacheExpiry
+        IdamCacheExpiry idamCacheExpiry
     ) {
         this.idamClient = idamClient;
         this.users = users;
-        this.accessTokenCache = Caffeine.newBuilder()
-            .expireAfter(accessTokenCacheExpiry)
-            .writer(new CacheWriter<String, CachedIdamToken>() {
-                @Override
-                public void write(@NonNull String key, @NonNull CachedIdamToken value) {
-                    throw new UnsupportedOperationException("Cache put() or replace() not supported.");
-                }
-
-                @Override
-                public void delete(
-                    @NonNull String jurisdiction,
-                    @NonNull CachedIdamToken cachedIdamToken,
-                    @NonNull RemovalCause cause
-                ) {
-                    log.info("Access token deleted for jurisdiction: {}, cause: {}, Invalidate user details.",
-                        jurisdiction,
-                        cause
-                    );
-                    userDetailsCache.invalidate(cachedIdamToken.accessToken);
-                }
-            })
-            .build();
-
-        this.userDetailsCache = Caffeine.newBuilder()
-            .maximumSize(200)
+        idamCache = Caffeine.newBuilder()
+            .expireAfter(idamCacheExpiry)
             .build();
     }
 
-    public String getAccessToken(String jurisdiction) {
-        log.info("Get access token for jurisdiction: {} ", jurisdiction);
-        CachedIdamToken cachedIdamToken = this.accessTokenCache
-            .get(jurisdiction, j -> retrieveToken(j));
-        return cachedIdamToken.accessToken;
+    public CachedIdamCredential getIdamCredentials(String jurisdiction) {
+        log.info("Get idam credential for jurisdiction: {} ", jurisdiction);
+        CachedIdamCredential cachedIdamCredential = idamCache.get(jurisdiction, this::retrieveIdamInfo);
+        return cachedIdamCredential;
     }
 
     public void removeAccessTokenFromCache(String jurisdiction) {
-        log.info("Remove access token from cache for jurisdiction: {} ", jurisdiction);
-        accessTokenCache.invalidate(jurisdiction);
+        log.info("Remove idam credential from cache for jurisdiction: {} ", jurisdiction);
+        idamCache.invalidate(jurisdiction);
     }
 
-    private CachedIdamToken retrieveToken(String jurisdiction) {
-        log.info("Retrieve access token for jurisdiction: {} from IDAM", jurisdiction);
+    private CachedIdamCredential retrieveIdamInfo(String jurisdiction) {
+        log.info("Retrieving access token for jurisdiction: {} from IDAM", jurisdiction);
```

Comment:
I like `retrieving...`, as it leaves no doubt that we're only about to do it. Maybe we could also apply it in the other logs, where we currently have `get` and `remove`?

Reply:
added